src/components/Canvas/Images.vue

```diff
@@ -7,433 +7,451 @@ Redistribution and use in source and binary forms, with or without modification,
-                    .domain([h, 0]) // inverted y axis
-                    .range([this.view.min_y, this.view.max_y])
-                    .invert(rwu);
+      anchor.on('mouseout', () => {
+        document.body.style.cursor = 'default';
+        anchor.setRadius(3)
+        .draw();
+      });
+
+      group.add(anchor);
+    },
+    /*
+    * Set the scale and position of the canvas based on the current viewbox
+    */
+    scaleAndPlaceStage() {
+      // original rwu/px resolution when scales were set
+      const originalResolution = (this.scaleX.range()[1] - this.scaleX.range()[0]) / (this.scaleX.domain()[1] - this.scaleX.domain()[0]);
+      // current rwu/px resolution based on current viewbox (after panning and zooming)
+      const currentResolution = (this.view.max_x - this.view.min_x) / this.$refs.images.clientWidth;
+      // scaling factor to render canvas images at current resolution
+      const scale = originalResolution / currentResolution;
+
+      this.stage
+        // scale will not be 1 if the user has zoomed in or out
+        .scale({
+          x: scale,
+          y: scale,
+        })
+        // place the canvas at the pixel value corresponding to our RWU 0
+        .position({
+          x: this.rwuToPx(0, 'x'),
+          y: this.rwuToPx(0, 'y'),
+        })
+        .draw();
+    },
+    rwuToPx(rwu, axis) {
+      let currentScale;
+      if (axis === 'x') {
+        currentScale = d3.scaleLinear()
+          .range([0, this.$refs.images.clientWidth])
+          .domain([this.view.min_x, this.view.max_x]);
+      } else if (axis === 'y') {
+        currentScale = d3.scaleLinear()
+          .range([this.$refs.images.clientHeight, 0]) // inverted y axis
+          .domain([this.view.min_y, this.view.max_y]);
+      }
+      return currentScale(rwu);
+    },
+  },
+  computed: {
+    ...mapState({
+      currentTool: state => state.application.currentSelections.tool,
+      images: state => state.application.currentSelections.story.images,
+      scaleX: state => state.application.scale.x,
+      scaleY: state => state.application.scale.y,
+      view: state => state.project.view,
+    }),
+
+    currentImage: {
+      get() { return this.$store.state.application.currentSelections.image; },
+      set(image) { this.$store.dispatch('application/setCurrentImage', { image }); },
+    },
+  },
+  watch: {
+    /*
+    * Compare the new images to a cached copy of the images because Vue will not store the previous value of an object that has changed
+    * konva will handle updating the canvas when the dimensions or location of an image are changed so don't rerender if
+    * if 'height', 'width', 'x', or 'y' have been updated
+    * if z index or opacity has changed, or an image has been added/removed, we must re render the canvas
+    */
+    images: {
+      handler(newImages) {
+        // an image has been added or removed
+        if (newImages.length !== this.imageCache.length) { this.renderImages(); }
+
```

Comment:
do you mean to have an `else` or a return here? Otherwise I can see 'renderImages' getting called twice.